src/Utils.jl

```diff
@@ -155,4 +155,6 @@ function _add_idmap_to_call(def::Expr, id_map::Expr)
     return Expr(:call, def.args[1], def.args[2:end]..., id_map)
 end
 
+@inline fill_similar(value, array, args...) = fill!(similar(array, args...), value)
```

Comment:
My one concern here is that some packages, when overloading `fill!`, will not return anything. Thus, this should create an array and then return it instead.

Reply:
/wtd Can you make this function explicitly return an array (which has been `fill!`'d with `value), rather than implicit as is done here

Reply:
/wtd Can you make this function explicitly return an array (which has been fill!'d with `value), rather than implicit as is done here